save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/Main.kt

```diff
@@ -53,23 +52,18 @@ internal val fs = FileSystem.SYSTEM
 
 @OptIn(ExperimentalSerializationApi::class)
 fun main() {
-    val configFromFile: AgentConfiguration = Properties.decodeFromStringMap(
-        readProperties("agent.properties")
-    )
-    // override with values from env
-    val config: AgentConfiguration = configFromFile.copy(
-        save = configFromFile.save.copy(
-            batchSize = optionalEnv(AgentEnvName.BATCH_SIZE)?.toInt(),
-            batchSeparator = optionalEnv(AgentEnvName.BATCH_SEPARATOR),
-            overrideExecCmd = optionalEnv(AgentEnvName.OVERRIDE_EXEC_CMD),
-            overrideExecFlags = optionalEnv(AgentEnvName.OVERRIDE_EXEC_FLAGS),
+    val propertiesFile = "agent.properties".toPath()
+    val config: AgentConfiguration = if (fs.exists(propertiesFile)) {
+        Properties.decodeFromStringMap(
+            readProperties("agent.properties")
         )
-    )
+    } else {
+        AgentConfiguration.initializeFromEnv()
```

Comment:
why we need  `AgentConfiguration.initializeFromEnv()` if we override it in `.updateFromEnv()`?

Reply:
I will change `requiredEnv` to `optionalEnv` in `updateFromEnv`, then it will possibly override properties from env